src/main/java/ladderGame/model/Name.java

```diff
@@ -12,15 +16,41 @@ public Name(String name) {
         this.name = name;
     }
 
+    public Name(String name, Players players) {
+        if(!name.equals("all")) {
+            validate(new Name(name), players);
+        }
+
+        this.name = name;
+    }
+
     private void validate(String name) {
         if (name.length() > MAXIMUM_NAME_LENGTH) {
             throw new IllegalArgumentException(EXCEPTION_MESSAGE_OVER_THAN_MAXIMUM);
         }
-        if (name.isBlank()) {
-            throw new IllegalArgumentException(EXCEPTION_MESSAGE_NOT_BLANK);
+        if(!NAME_PATTERN.matcher(name).matches()) {
+            throw new IllegalArgumentException(EXCEPTION_MESSAGE_UNDEFINED_NAME);
         }
     }
 
+    private void validate(Name name, Players players) {
+        if(!players.contains(name)) {
+            throw new IllegalArgumentException(EXCEPTION_MESSAGE_NON_CONTAIN_NAME);
+        }
+    }
+
+    public boolean isAll() {
+        return name.equals("all");
+    }
```

Comment:
```suggestion
    public boolean isAll() {
        return "all".equals(name);
    }
```
가능성은 없겠지만 항상 equals비교할때는 항상 상수가 먼저오도록 작성해주세요
그리고 all은 상수로 빼면 좋을것 같네요